src/installer.ts

```diff
@@ -85,7 +85,14 @@ async function isInstalled(
       .then(() => p)
       .catch(() => undefined);
 
-    if (installedPath) return success(tool, version, installedPath);
+    if (installedPath) {
+      // Make sure that the correct ghc is used, even if ghcup has set a
+      // default prior to this action being ran.
+      if (tool === 'ghc' && installedPath === ghcupPath)
```

Comment:
I suspect it's just macOS currently, but this problem would also show up on any system where `ghcup set` was called. Wouldn't be terribly common, but I could envision a pipeline where someone used setup-haskell twice rather than a matrix? Either way, it's best to be robust and fix the actual underlying issue I think :)